docs/plugins/plugin-development-guide/implementation.rst

```diff
@@ -0,0 +1,179 @@
+Implementation
+--------------
+
+The goal of our plugin is simple - we need to extend ``ProductVariant`` entity and provide new flag, that could be set
+on product variant form. Following customizations uses information provided in **Sylius Customization Guide**, especially
+from :doc:`customizing models</customization/model>`, :doc:`form</customization/form>` and :doc:`template</customization/template>`
+pages.
+
+Model
+*****
+
+The only field we need to add is additional ``$availableOnDemand`` boolean. Of course we should start with the unit tests (written with
+PHPSpec/PHPUnit/any other unit testing tool):
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace spec\IronMan\SyliusProductOnDemandPlugin\Entity;
+
+    use IronMan\SyliusProductOnDemandPlugin\Entity\ProductVariantInterface;
+    use PhpSpec\ObjectBehavior;
+    use Sylius\Component\Core\Model\ProductVariant;
+
+    final class ProductVariantSpec extends ObjectBehavior
+    {
+        function it_is_sylius_product_variant(): void
+        {
+            $this->shouldHaveType(ProductVariant::class);
+        }
+
+        function it_implements_product_variant_interface(): void
+        {
+            $this->shouldImplement(ProductVariantInterface::class);
+        }
+
+        function it_can_be_available_on_demand(): void
+        {
+            $this->setAvailableOnDemand(true);
+            $this->isAvailableOnDemand()->shouldReturn(true);
+        }
+    }
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace IronMan\SyliusProductOnDemandPlugin\Entity;
+
+    use Sylius\Component\Core\Model\ProductVariant as BaseProductVariant;
+
+    class ProductVariant extends BaseProductVariant implements ProductVariantInterface
+    {
+        /** @var bool */
+        private $availableOnDemand;
+
+        public function setAvailableOnDemand(?bool $availableOnDemand): void
+        {
+            $this->availableOnDemand = $availableOnDemand;
+        }
+
+        public function isAvailableOnDemand(): ?bool
+        {
+            return $this->availableOnDemand;
+        }
+    }
+
+.. important::
+
+    Of course you need to remember about entity mapping customization as well.
```

Comment:
I think we should add example here, it will be really simple but important and easy to miss if you are new to Sylius.